docs/proposals/workflow.md

```diff
@@ -0,0 +1,134 @@
+
+    // Spec is a structure defining the expected behavior of a job.
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+    Spec JobSpec `json:"spec,omitempty"`
+
+    // Status is a structure describing current status of a job.
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+    Status JobStatus `json:"status,omitempty"`
+
+    // JobSelector to detect parents jobs
+    ParentSelector *JobSelector `json:"parentSelector, omitempty"`
+}
+```
+
+#### JobSpec and JobStatus
+
+No modifications
+
+#### Parent JobSelector
+
+The `.job.parentselector` is a label query over a set of jobs.
+If all selected jobs are completed the `job` will be started immediately.
+If `.job.parentselector` is absent the `job` will be started immediately.
+Since labels are forwarded directly from `job` to pods, users should not create pods whose labels match this selector, either directly,
+via another Job, or via another controller (for example Replication controller), [see](https://github.com/kubernetes/kubernetes/issues/14961).
+
+#### Some special cases
+
+Jobs terminated via [#17244](https://github.com/kubernetes/kubernetes/issues/17244)  (if implemented) will be considered _completed_ so will trigger the _child_ jobs. Since it's very unlikely that the user would keep the _child_ jobs alive we may add a field to cascade termination to _child_ jobs if any to the current proposals made by @pmorie and @bgrant0607.
+
+For example
+
+```json
+apiVersion: v1alpha1
+kind: UpwardAPIRequest
+spec:
+    terminateExistingJob: true
+    reason: "TimeOut"
+    message: "Pods ran out of time"
+    cascade: true
+```
+
+### CRUD
+
+    - Creating a job without an empty `jobs.parentselector` will simply follow the usual `job` life cycle.  With a non non nil `jobs.parentselector` the logic already described will be considered.
+    - Users may read the `job` using the usual `get`, `describe` commands already implemented for `job`. Obviously the `describe` command should display the non nil `jobs.parentsselector` (if any).
+    - A `job` cannot be updated. To update a `job`, user should delete and re-create it. The only exception is `job` _scaling_. A `workflow job` (i.e. with a non nil selector) can be scaled in the common way. No matter if pods are currently running or not.
+    - A `job` can be deleted in the usual way `job`. A cascade mechanism may be implemented in a later phase.
+
+## Events
+
+The usual Job controller events will be emitted.
+    * JobStart
+    * JobFinished
+Since a Job with a non nil job selector can be created and may never start we propose to add the event:
+    * JobCreated
+
+## Known drawbacks
+
+    * Using only a label selector won't permit to implement backtracking for failures, a common functionality for a DAG worflow system. In this cases [controllerRef](https://github.com/kubernetes/kubernetes/issues/2210#issuecomment-134878215) could help.
```

Comment:
For backtracking I mean the ability to backtrack from a failed `job` to its parent job (which most of the time is the reason for the failure). Our current tool has this troubleshooting ability since the causality of the job is very clear. Using `labels selector` and `boolean field` only won't give us this feature.